### ISSUE TITLE
PickleCorpusView raising UnicodeDecodeError when reading file

### ISSUE THREAD

**Issue by alexandredias3d** [nltk/nltk#2331]
I have been trying to save a modified corpus produced with `LazyMap` using`PickleCorpusView.write` as shown in the [PickleCorpusView documentation](https://www.nltk.org/_modules/nltk/corpus/reader/util.html#PickleCorpusView).

The writing seems to be working properly. However, when trying to read the pickled file, it raises a `UnicodeDecodeError`.

The following code does not even use a `LazyMap` but raises the same error

```python
Python 3.7.3 (default, Jun 24 2019, 04:54:02) 
[GCC 9.1.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import nltk
>>> nltk.corpus.reader.PickleCorpusView.write('In the real use case, this would be the result of LazyMap producing a list of list of tuples (word and PoS tag).', 'test.pickle')
>>> nltk.corpus.reader.PickleCorpusView('test.pickle')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/user/.virtualenv/venv/lib/python3.7/site-packages/nltk/collections.py", line 231, in __repr__
    for elt in self:
  File "/home/user/.virtualenv/venv/lib/python3.7/site-packages/nltk/corpus/reader/util.py", line 306, in iterate_from
    tokens = self.read_block(self._stream)
  File "/home/user/.virtualenv/venv/lib/python3.7/site-packages/nltk/corpus/reader/util.py", line 525, in read_block
    result.append(pickle.load(stream))
  File "/home/user/.virtualenv/venv/lib/python3.7/site-packages/nltk/data.py", line 1177, in read
    chars = self._read(size)
  File "/home/user/.virtualenv/venv/lib/python3.7/site-packages/nltk/data.py", line 1469, in _read
    chars, bytes_decoded = self._incr_decode(bytes)
  File "/home/user/.virtualenv/venv/lib/python3.7/site-packages/nltk/data.py", line 1491, in _incr_decode
    return self.decode(bytes, 'strict')
  File "/home/user/.virtualenv/venv/lib/python3.7/encodings/utf_8.py", line 16, in decode
    return codecs.utf_8_decode(input, errors, True)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0x80 in position 0: invalid start byte
```

Am I missing something from the documentation?

I am using Python 3.7.3 and NLTK 3.4.1 running on Arch Linux (kernel 5.1.15). The same error also occurred in Ubuntu.

**Comment by alvations** [comment]
@alexandredias3d Thank you for raising the issue. Could you share the `test.pickle` file and explain what's inside so that we can look into the problem?

**Comment by DangMinh24** [comment]
@alexandredias3d I have the same problem. Turn out my pickled files is not in format of 'utf-8'. And `PickleCorpusView` default open the pickled file in format 'utf-8'. I change the source code of  `PickleCorpusView` from

```python
class PickleCorpusView(StreamBackedCorpusView):

    BLOCK_SIZE = 100
    PROTOCOL = -1

    def __init__(self, fileid, delete_on_gc=False):
        self._delete_on_gc = delete_on_gc
        StreamBackedCorpusView.__init__(self, fileid)
````
with `encoding=None` at the `__init__`
```python
class PickleCorpusView(StreamBackedCorpusView):

    BLOCK_SIZE = 100
    PROTOCOL = -1

    def __init__(self, fileid, delete_on_gc=False):
        self._delete_on_gc = delete_on_gc
        StreamBackedCorpusView.__init__(self, fileid, encoding=None)
```

**Comment by stevenbird** [comment]
I believe this will be resolved in NLTK 3.5
